test/test_threads.jl

```diff
@@ -23,20 +26,21 @@ const N = 10
 dst = mod(rank+1, size)
 src = mod(rank-1, size)
 
-if provided == MPI.THREAD_MULTIPLE
-    send_arr = collect(1.0:N)
-    recv_arr = zeros(N)
+# DEBUG: currently failing on UCX error
```

Comment:
can you mask it out by `if ArrayType != Array`?

Reply:
Shouldn't it be `if ArrayType==Array` (then execute the `if provided == MPI.THREAD_MULTIPLE` block - since it fails `if ArrayType==ROCArray`)

Reply:
I can mask it out, wrapping the `if provided == MPI.THREAD_MULTIPLE [...] end` into the suggested `if`. But then the test still execute for `ArrayType==ROCArray` which leads to a segfault (see [error_rocarray.txt](https://github.com/JuliaParallel/MPI.jl/files/8484887/error_rocarray.txt)).

Also, if using `ArrayType==Array` for testing, then the test errors as well (see [error_array.txt](https://github.com/JuliaParallel/MPI.jl/files/8484888/error_array.txt)).

Reply:
@simonbyrne Does the error stack tell you something what to search for? What to do as next steps. Seems the failure is not related to ROCm addition since it fails also with `Array`s. Happy to work through the steps needed to be able to merge the ROCm additions 🙂